pkg/controller/controller_utils_test.go

```diff
@@ -376,31 +376,64 @@ func TestDeletePodsAllowsMissing(t *testing.T) {
 	assert.True(t, apierrors.IsNotFound(err))
 }
 
+func TestCountTerminatingPods(t *testing.T) {
+	now := metav1.Now()
+
+	// This rc is not needed by the test, only the newPodList to give the pods labels/a namespace.
+	rc := newReplicationController(0)
+	podList := newPodList(nil, 7, v1.PodRunning, rc)
+	podList.Items[0].Status.Phase = v1.PodSucceeded
+	podList.Items[1].Status.Phase = v1.PodFailed
+	podList.Items[2].Status.Phase = v1.PodPending
+	podList.Items[2].SetDeletionTimestamp(&now)
+	podList.Items[3].Status.Phase = v1.PodRunning
+	podList.Items[3].SetDeletionTimestamp(&now)
+	var podPointers []*v1.Pod
+	for i := range podList.Items {
+		podPointers = append(podPointers, &podList.Items[i])
+	}
+
+	terminatingPods := CountTerminatingPods(podPointers)
+
+	assert.Equal(t, pointer.Int32Deref(terminatingPods, 0), int32(2))
+
+	terminatingList := FilterTerminatingPods(podPointers)
+	assert.Equal(t, len(terminatingList), int(2))
+}
 func TestActivePodFiltering(t *testing.T) {
+	now := metav1.Now()
+
 	// This rc is not needed by the test, only the newPodList to give the pods labels/a namespace.
 	rc := newReplicationController(0)
-	podList := newPodList(nil, 5, v1.PodRunning, rc)
+	podList := newPodList(nil, 7, v1.PodRunning, rc)
 	podList.Items[0].Status.Phase = v1.PodSucceeded
 	podList.Items[1].Status.Phase = v1.PodFailed
-	expectedNames := sets.NewString()
-	for _, pod := range podList.Items[2:] {
-		expectedNames.Insert(pod.Name)
+	podList.Items[2].Status.Phase = v1.PodPending
+	podList.Items[2].SetDeletionTimestamp(&now)
+	podList.Items[3].Status.Phase = v1.PodRunning
+	podList.Items[3].SetDeletionTimestamp(&now)
+
+	expectedNamesNoTerminating := sets.NewString()
+	for _, pod := range podList.Items[4:] {
+		expectedNamesNoTerminating.Insert(pod.Name)
 	}
 
 	var podPointers []*v1.Pod
 	for i := range podList.Items {
 		podPointers = append(podPointers, &podList.Items[i])
 	}
+
 	got := FilterActivePods(podPointers)
 	gotNames := sets.NewString()
 	for _, pod := range got {
 		gotNames.Insert(pod.Name)
 	}
 
-	assert.Equal(t, 0, expectedNames.Difference(gotNames).Len(),
-		"expected %v, got %v", expectedNames.List(), gotNames.List())
-	assert.Equal(t, 0, gotNames.Difference(expectedNames).Len(),
-		"expected %v, got %v", expectedNames.List(), gotNames.List())
+	assert.Equal(t, 0, expectedNamesNoTerminating.Difference(gotNames).Len(),
```

Comment:
Prefer `cmp.Diff`

Reply:
IMO I really like to follow the pattern of adopting the style of the file I am in.  

It looks like every other test case here uses `Difference`.  

This file could use a little love as no one of the tests are table driven and they all use `Difference` instead of `cmp.Diff`. 

I don't know if I agree with making this change in this PR though.

Reply:
Unfortunately, the way these tests are written don't output the difference in the error message.

But I agree that this is beyond the scope of this PR.

Reply:
I'll create an issue.

Reply:
https://github.com/kubernetes/kubernetes/issues/119133 is created.

Reply:
Looks like that issue was picked up so I’ll resolve this discussion.